docs/topics/coroutines-overview.md

```diff
@@ -1,39 +1,75 @@
-- [Asynchronous flow](flow.md)
+> Check out the [KotlinConf app](https://github.com/JetBrains/kotlinconf-app) for a sample project to see how coroutines are used in practice.
+> 
+{style="tip"}
 
-### Tutorials
+## Coroutine concepts
 
-- [Asynchronous programming techniques](async-programming.md)
-- [Introduction to coroutines and channels](coroutines-and-channels.md)
-- [Debug coroutines using IntelliJ IDEA](debug-coroutines-with-idea.md)
-- [Debug Kotlin Flow using IntelliJ IDEA – tutorial](debug-flow-with-idea.md)
-- [Testing Kotlin coroutines on Android](https://developer.android.com/kotlin/coroutines/test)
+The `kotlinx.coroutines` library provides the core building blocks for running tasks concurrently, structuring coroutine execution, and managing shared state.
 
-## Sample projects
+### Suspending functions and coroutine builders
 
-- [kotlinx.coroutines examples and sources](https://github.com/Kotlin/kotlin-coroutines/tree/master/examples)
-- [KotlinConf app](https://github.com/JetBrains/kotlinconf-app)
+Coroutines in Kotlin are built on suspending functions, which allow code to pause and resume without blocking a thread.
+The `suspend` keyword marks functions that can perform long-running operations asynchronously.
 
+To launch new coroutines, use coroutine builders like [`.launch()`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/launch.html) and [`.async()`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/async.html).
+These builders are extension functions on [`CoroutineScope`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-scope/),
+which defines the coroutine’s lifecycle and provides the coroutine context.
 
+You can learn more about these builders in [Coroutine basics](coroutine-basics.md) and [Composing suspend functions](coroutines-and-channels.md).
+
+### Coroutine context and behavior
+
+Launching a coroutine from a `CoroutineScope` creates a context that governs its execution.
+Builder functions like `.launch()` and `.async()` automatically create a set of elements that define how the coroutine behaves:
+
+* The [`Job`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-job/) interface tracks the coroutine's lifecycle and enables structured concurrency.
+* [`CoroutineDispatcher`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-dispatcher/) controls where the coroutine runs, such as on a background thread or the main thread in UI applications.
+* [`CoroutineExceptionHandler`](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/-coroutine-exception-handler/) handles uncaught exceptions.
+
+Together, these elements make up the [_coroutine context_](coroutine-context-and-dispatchers.md), which is inherited by default from the coroutine's parent.
```

Comment:
There are more potential context elements, though, these are only the main ones. To avoid misleading, some form of "and others" should be introduced.